Squeegee/minimal_squeegee_runoff_drainage.py

```diff
@@ -0,0 +1,293 @@
+        facets.cbar.set_label("Averaging weights")
+        plt.savefig('weights.png')
+
+    return TSr, savg, resultn
+
+#store the dfs in dict
+
+# here the sparse matrix with grid cell weights for each ploygon is generated
+# see xesmf doc: 
+# https://pangeo-xesmf.readthedocs.io/en/latest/notebooks/Spatial_Averaging.html
+# the matrix can be stored as a nc-file and re-used, so that the program runs 
+# very fast once it is generated. 
+# The weights are specific to the domain, projection, resolution, and to 
+# options like 
+#&nam_pgd_arrange_cover 
+#  lwater_to_nature = .true. 
+#  ltown_to_rock = .true.
+#  which alters where RUNOFFC etc is defined 
+
+
+def run(dictds, expname, TSr, savg, resultn):
+
+    numruns=len(expnam)
+    cachrunoff={}
+
+    for i in np.arange(0,numruns):
+
+        tmp1=dictds[expnam[i]]['RUNOFFC_ISBA'] +dictds[expnam[i]]['DRAINC_ISBA']
+    
+        #some experiments have accumlated runoff (i.e. do not use option
+        # LRESETCUMUL = .true. in NAM_WRITE_DIAG_SURFn 
+        # see https://www.umr-cnrm.fr/surfex/spip.php?article406 )
+        # so need to deaccumulate
+        if expnam[i] in ['nm12snowl', 'nmdtB07','nmSOC','nbni','s']:
+            tmp1=tmp1.diff(dim='time')
+        tmp1=tmp1.to_dataset(name=expnam[i])
+        tmp1['mask']=TSr.RUNOFFC_ISBA.isnull()
+        tmp1['mask'].values=np.where(~TSr.RUNOFFC_ISBA.isnull(),1,0)
+        out=savg(tmp1[expnam[i]])
+        #out = out.assign_coords(roms_id=xr.DataArray(resultn.index.values, dims=("roms_id",)))
+        out = out.assign_coords(stID=xr.DataArray(resultn["stID"], dims=("stID",)))
+        print("out")
+        print(out)
+        cachrunoff[expnam[i]]=out.to_pandas()
+        rainfRun= cachrunoff[expnam[i]]=out.to_pandas()
+        print("rainfRun")
+        print(rainfRun)
+        rainfRun["50.64.0"].plot()
+        plt.show()
+        cachrunoff[expnam[i]].to_csv(expnam[i]+'.csv')
+
+
+if __name__ == "__main__":
+
+    expdirs='/lustre/storeB/project/nwp/H2O/wp4/SURFEX_offline/open_SURFEX_V8_1/MY_RUN/KTEST/'
+    forcingfile = '/lustre/storeB/project/nwp/H2O/wp4/FORCING/527_450_2020/FORCING_527_450_202010.nc'
+    stationfile = '/lustre/storeB/project/nwp/H2O/wp4/RRdata/stations_Huang20.txt'
+#    shapefiles = '/lustre/storeB/project/nwp/H2O/wp4/RRdata/shapefiles/CCM2/ccm21/WGS84_W2008.gdb'
+    shapefiles = '/lustre/storeB/project/nwp/H2O/wp4/RRdata/shapefiles/latlonHyd/latlonhydorder/NVEData/Hydrologi/Hydrologi_TotalNedborfeltMalestasjon.shp'
+    weightfile = '/lustre/storeB/users/josteinbl/TOPD/spatial_avg_4catchments_527nature2lim.nc'
+
+    # folder where runs are, this could have been a long list of different exps.
+    expdirfs=['527_450/']
+    # start of nc-file-output from runs
+    ncstart=  ['openofl']
+    # choose a name for legend in plots
+    expnam=['mebglac_albevol']
+    variables = ['RUNOFFC_ISBA', 'DRAINC_ISBA']
+
+    myP, proj_string, fPREP, fPGD, FORCING = getDomainData(expdirs)
```

Comment:
Get the domain from a domain definition. You can also use pysurfex, or add it into pysurfex